test/python/test_onedim.py

```diff
@@ -1263,6 +1288,45 @@ def test_bad_boundary_conditions(self):
         sim.fuel_inlet.X = "H2: 1.0"
         sim.set_initial_guess()
 
+    def run_restore_diffusionflame(self, fname):
+        gas = ct.Solution("h2o2.yaml")
+        sim = ct.CounterflowDiffusionFlame(gas)
+        sim.restore(fname)
+        radial_lambda = sim.flame.radial_pressure_gradient
+        assert np.isclose(np.unique(radial_lambda)[0], -90257.82873678)
+        assert np.allclose(sim.T, sim.flame.temperature)
+        assert np.allclose(radial_lambda, sim.flame.L)
+        assert np.allclose(radial_lambda, sim.flame.Lambda)
+        assert np.allclose(radial_lambda, getattr(sim.flame, "lambda"))
+        zeros = sim.T * 0
+        with pytest.raises(ct.CanteraError, match="is not used by 'axisymmetric-flow'"):
+            sim.flame.set_values("electric-field", zeros)
+        with pytest.raises(ct.CanteraError, match="is not used by 'axisymmetric-flow'"):
+            sim.flame.set_values("oxidizer-velocity", zeros)
+
+    def run_restore_solutionarray(self, fname):
+        gas = ct.Solution("h2o2.yaml")
+        arr = ct.SolutionArray(gas)
+        arr.restore(fname, "solution/flame")
+
+        radial_lambda = np.unique(arr.radial_pressure_gradient)
+        assert len(radial_lambda) == 1
+        assert radial_lambda[0] == pytest.approx(-90257.82873678)
+        assert getattr(arr, "lambda")[0] == radial_lambda[0]
+        assert radial_lambda[0] == arr.L[0]
+        assert radial_lambda[0] == arr.Lambda[0]
+        assert getattr(arr, "radial-pressure-gradient")[0] == radial_lambda[0]
+
+    def test_restore_lambda_yaml(self):
+        self.run_restore_diffusionflame("flame_lambda.yaml")
+        self.run_restore_solutionarray("flame_lambda.yaml")
+
+    @pytest.mark.skipif("native" not in ct.hdf_support(),
+                        reason="Cantera compiled without HDF support")
+    def test_restore_lambda_hdf(self):
+        self.run_restore_diffusionflame("flame_lambda.h5")
+        self.run_restore_solutionarray("flame_lambda.h5")
```

Comment:
Rather than modifying this to consider the input file search path, you can use the `test_data_path` fixture to access files in `test/data`.

Also, as this is one of the larger files in `test/data`, I'd suggest generating it from a simulation without grid refinement enabled to reduce the size a bit.

Reply:
See above.